### PR TITLE
Doc release guide

### PR DESCRIPTION
**Pull request by tacaswell**
Plan to fill this out as I go for the 1.5.2 release.

### CODE REVIEW

File: doc/devel/release_guide.rst [hunk 1]
```diff
@@ -6,27 +6,98 @@ Doing a matplotlib release
...
-* Run :file:`unit/memleak_hawaii3.py` and make sure there are no
-  memory leaks
+   python tests.py --processes=8 --process-timeout=300
 
-* try some GUI examples, e.g., :file:`simple_plot.py` with GTKAgg, TkAgg, etc...
+In addition ::
 
-* remove font cache and tex cache from :file:`.matplotlib` and test
-  with and without cache on some example script
+   python unit/memleak_hawaii3.py
```
Comment (QuLogic) [inline doc/devel/release_guide.rst]: Should we be running on Python 3 nowadays?

Comment (tacaswell) [inline doc/devel/release_guide.rst]: Good point.  I normally work in a conda env so python is python3 (and most of my machines are arch so python is also python3) so I forget that is not universally the case.

**Comment by jenshnielsen** [comment]
Since we haven't merged #6044 of fixed #4205 perhaps it's useful to do something like 
```
linkchecker build/html/index.html --check-extern
```
or using the build in linkchecer in Sphinx to check that our external links are not broken.

File: doc/devel/release_guide.rst [hunk 2]
```diff
@@ -6,27 +6,98 @@ Doing a matplotlib release
...
+============
+
+To make sure everyone's hard work gets credited, regenerate the github
+stats.  In the project root run ::
+
+  python tools/github_stats.py --since-tag $TAG --project 'matplotlib/matplotlib' --links > doc/users/github_stats.rst
+
+
+where `$TAG` is the tag of the last major release.  This will generate
+stats for all work done since that release.
```
Comment (tacaswell) [inline doc/devel/release_guide.rst]: add note about installing requests + requests_cache

File: doc/devel/release_guide.rst [hunk 3]
```diff
@@ -6,27 +6,98 @@ Doing a matplotlib release
...
+
+.. _release_ghstats:
+
+Github Stats
+============
+
+To make sure everyone's hard work gets credited, regenerate the github
+stats.  In the project root run ::
+
+  python tools/github_stats.py --since-tag $TAG --project 'matplotlib/matplotlib' --links > doc/users/github_stats.rst
```
Comment (tacaswell) [inline doc/devel/release_guide.rst]: add note about editing to use pre-existing oath token

**Pull request closed without merging**